### ISSUE TITLE
Trouble Connecting to Debugger (Insists on Using localhost?)

### ISSUE THREAD

**Issue by jamiejackson** [softwareCobbler/luceedebug#10]
I installed the extension via the Marketplace. :-)

However, I'm getting the following. It seems like it's trying `localhost`, instead of the host I supplied.

<img width="742" alt="image" src="https://user-images.githubusercontent.com/479475/206919794-e125fffa-0764-40f2-9edf-6ffbe529ee51.png">

```json
{
    // Use IntelliSense to learn about possible attributes.
    // Hover to view descriptions of existing attributes.
    // For more information, visit: https://go.microsoft.com/fwlink/?linkid=830387
    "version": "0.2.0",
    "configurations": [
    //     {
    //         "type": "cfml-debugger",
    //         "request": "launch",
    //         "name": "CFML Debug",
    //         "program": "${workspaceFolder}/${command:AskForProgramName}",
    //         "globalConfigFilePath": "${command:globalConfigFilePath}",
    //         "project": "${command:AskForProject}",
    //         "stopOnEntry": false,
    //         "windows": {
    //             "program": "${workspaceFolder}\\${command:AskForProgramName}"
    //         }
    //     }

    {
        "type": "cfml",
        "request": "attach",
        "name": "Project A",
        "hostName": "local.hudx.net",
        "port": 10000,
        "pathTransform": { // optional
            "idePrefix": "${workspaceFolder}/cfml/deployment_root",
            "cfPrefix": "/app/www/"
        }
    }
    ]

}
```

As a sanity check, I then changed the port (to `10001`), which _is_ reflected in the popup:

<img width="261" alt="image" src="https://user-images.githubusercontent.com/479475/206919903-ca2a8344-c9b4-4cd8-a63b-9049effa4a48.png">

**Comment by softwareCobbler** [comment]
Fixed in 2.0.1, should be available in next few minutes (if you look at the extension via the sidebar it will eventually say something like "restart vscode to install latest version")

**Comment by jamiejackson** [comment]
That worked, thanks. However, I still can't connect:

<img width="262" alt="image" src="https://user-images.githubusercontent.com/479475/206924848-a421cdc0-1f09-429d-add0-5c0c52b62c4e.png">

My config:

```
-agentlib:jdwp=transport=dt_socket,server=y,suspend=n,address=localhost:9999
-javaagent:/app/lib/luceedebug/luceedebug.jar=jdwpHost=localhost,jdwpPort=9999,cfHost=0.0.0.0,cfPort=10000,jarPath=/app/lib/luceedebug/luceedebug.jar
```

I have tried with `cfHost` as `0.0.0.0` and `localhost`.

Do you spot anything wrong with my configuration or do you have any tips for how to debug the connection?

I'd like to verify that it's actually listening at 10000 but I don't know how to check for signs of life.

Should nmap give me anything useful?

```
% nmap local.hudx.net -p 443,10000
Starting Nmap 7.92 ( https://nmap.org ) at 2022-12-11 14:45 EST
Nmap scan report for local.hudx.net (192.168.56.10)
Host is up (0.013s latency).

PORT      STATE  SERVICE
443/tcp   open   https
10000/tcp closed snet-sensor-mgmt
```

The only debug-related port that I'm publishing outside of the container is 10000. The jdwp stuff: That's all internal to the container, right? I don't need to publish that port?

**Comment by softwareCobbler** [comment]
The jdwp stuff is internal to the container, that's right.

The "closed" state in the nmap output suggests the container is not binding the desired port.

In my case I have a docker compose file like
```
services:
  ...
  ports:
    - 10000 : 10000 # bind host machine port to container port
```

With that config, running your nmap command gives `port 10000 state: open`; but if I comment that port config line out, I get `state: closed` (and get the ECONNREFUSED errors)

**Comment by jamiejackson** [comment]
Unfortunately, I'm already publishing `10000` in the same way.

```
$ docker service list | grep hudx_local_lucee
rojbbg7pu2ll   hudx_local_lucee                      replicated   1/1        main-stack/lucee:local_20221211.122927                      *:8088->8088/tcp, *:8888->8888/tcp, *:10000->10000/tcp
```

**Comment by softwareCobbler** [comment]
Hm, a troubling case of "works on my machine". Not sure what a good next trouble shooting step would be.

All the network stuff gets booted up here, maybe strategic `println` could be helpful.
https://github.com/softwareCobbler/luceedebug/blob/6d2c3f3533f1a379133f19a67b66039d97221cbb/luceedebug/src/main/java/luceedebug/coreinject/DebugManager.java#L31

**Comment by jamiejackson** [comment]
Now I can see `cfHost`:`cfPort` "filtered," which is a good sign. However, the debugger still can't connect.

I hadn't checked the JDWP port before but I see that it's closed:

```
$ docker exec -it $(get_a_container_id_from_service.sh hudx_local_lucee) nmap localhost -p 8888,9999,10000
Starting Nmap 7.70 ( https://nmap.org ) at 2022-12-12 10:10 CST
Nmap scan report for localhost (127.0.0.1)
Host is up (0.000042s latency).
Other addresses for localhost (not scanned): ::1

PORT      STATE  SERVICE
8888/tcp  open   sun-answerbook
9999/tcp  closed abyss
10000/tcp open   snet-sensor-mgmt

Nmap done: 1 IP address (1 host up) scanned in 0.29 seconds
```

Is that to be expected or is that a potential clue?

Here are the startup logs, by the way:

```
Using CATALINA_BASE:   /usr/local/tomcat
Using CATALINA_HOME:   /usr/local/tomcat
Using CATALINA_TMPDIR: /usr/local/tomcat/temp
Using JRE_HOME:        /usr/local/openjdk-11
Using CLASSPATH:       /usr/local/tomcat/bin/bootstrap.jar:/usr/local/tomcat/bin/tomcat-juli.jar
Using CATALINA_OPTS:
NOTE: Picked up JDK_JAVA_OPTIONS:  --add-opens=java.base/java.lang=ALL-UNNAMED --add-opens=java.base/java.io=ALL-UNNAMED --add-opens=java.rmi/sun.rmi.transport=ALL-UNNAMED
Listening for transport dt_socket at address: 9999
[luceedebug] agent premain complete
12-Dec-2022 09:53:28.908 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log Server version name:   Apache Tomcat/9.0.39
12-Dec-2022 09:53:28.913 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log Server built:          Oct 6 2020 14:11:46 UTC
12-Dec-2022 09:53:28.913 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log Server version number: 9.0.39.0
12-Dec-2022 09:53:28.914 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log OS Name:               Linux
12-Dec-2022 09:53:28.921 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log OS Version:            4.15.0-200-generic
12-Dec-2022 09:53:28.922 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log Architecture:          amd64
12-Dec-2022 09:53:28.923 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log Java Home:             /usr/local/openjdk-11
12-Dec-2022 09:53:28.924 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log JVM Version:           11.0.9+11
12-Dec-2022 09:53:28.925 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log JVM Vendor:            Oracle Corporation
12-Dec-2022 09:53:28.925 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log CATALINA_BASE:         /usr/local/tomcat
12-Dec-2022 09:53:28.926 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log CATALINA_HOME:         /usr/local/tomcat
12-Dec-2022 09:53:28.936 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log Command line argument: --add-opens=java.base/java.lang=ALL-UNNAMED
12-Dec-2022 09:53:28.938 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log Command line argument: --add-opens=java.base/java.io=ALL-UNNAMED
12-Dec-2022 09:53:28.939 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log Command line argument: --add-opens=java.rmi/sun.rmi.transport=ALL-UNNAMED
12-Dec-2022 09:53:28.939 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log Command line argument: -Djava.util.logging.config.file=/usr/local/tomcat/conf/logging.properties
12-Dec-2022 09:53:28.940 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log Command line argument: -Djava.util.logging.manager=org.apache.juli.ClassLoaderLogManager
12-Dec-2022 09:53:28.941 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log Command line argument: -Xms512m
12-Dec-2022 09:53:28.941 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log Command line argument: -Xmx4096m
12-Dec-2022 09:53:28.942 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log Command line argument: -Djava.security.egd=file:/dev/./urandom
12-Dec-2022 09:53:28.942 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log Command line argument: -agentlib:jdwp=transport=dt_socket,server=y,suspend=n,address=localhost:9999
12-Dec-2022 09:53:28.950 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log Command line argument: -javaagent:/app/lib/luceedebug/luceedebug.jar=jdwpHost=localhost,jdwpPort=9999,cfHost=localhost,cfPort=10000,jarPath=/app/lib/luceedebug/luceedebug.jar
12-Dec-2022 09:53:28.950 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log Command line argument: -Djdk.tls.ephemeralDHKeySize=2048
12-Dec-2022 09:53:28.952 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log Command line argument: -Djava.protocol.handler.pkgs=org.apache.catalina.webresources
12-Dec-2022 09:53:28.953 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log Command line argument: -Dorg.apache.catalina.security.SecurityListener.UMASK=0027
12-Dec-2022 09:53:28.953 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log Command line argument: -Dignore.endorsed.dirs=
12-Dec-2022 09:53:28.954 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log Command line argument: -Dcatalina.base=/usr/local/tomcat
12-Dec-2022 09:53:28.954 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log Command line argument: -Dcatalina.home=/usr/local/tomcat
12-Dec-2022 09:53:28.955 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log Command line argument: -Djava.io.tmpdir=/usr/local/tomcat/temp
12-Dec-2022 09:53:28.964 INFO [main] org.apache.catalina.core.AprLifecycleListener.lifecycleEvent Loaded Apache Tomcat Native library [1.2.25] using APR version [1.6.5].
12-Dec-2022 09:53:28.964 INFO [main] org.apache.catalina.core.AprLifecycleListener.lifecycleEvent APR capabilities: IPv6 [true], sendfile [true], accept filters [false], random [true].
12-Dec-2022 09:53:28.965 INFO [main] org.apache.catalina.core.AprLifecycleListener.lifecycleEvent APR/OpenSSL configuration: useAprConnector [false], useOpenSSL [true]
12-Dec-2022 09:53:28.971 INFO [main] org.apache.catalina.core.AprLifecycleListener.initializeSSL OpenSSL successfully initialized [OpenSSL 1.1.1d  10 Sep 2019]
12-Dec-2022 09:53:29.504 INFO [main] org.apache.coyote.AbstractProtocol.init Initializing ProtocolHandler ["http-nio-8888"]
12-Dec-2022 09:53:29.563 INFO [main] org.apache.catalina.startup.Catalina.load Server initialization in [896] milliseconds
12-Dec-2022 09:53:29.627 INFO [main] org.apache.catalina.core.StandardService.startInternal Starting service [Catalina]
12-Dec-2022 09:53:29.627 INFO [main] org.apache.catalina.core.StandardEngine.startInternal Starting Servlet engine: [Apache Tomcat/9.0.39]
WARNING: An illegal reflective access operation has occurred
WARNING: Illegal reflective access by org.apache.felix.framework.ext.ClassPathExtenderFactory$DefaultClassLoaderExtender (file:/usr/local/tomcat/lucee/lucee.jar) to method java.net.URLClassLoader.addURL(java.net.URL)
WARNING: Please consider reporting this to the maintainers of org.apache.felix.framework.ext.ClassPathExtenderFactory$DefaultClassLoaderExtender
WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
WARNING: All illegal access operations will be denied in a future release
log4j:WARN No appenders could be found for logger (org.jets3t.service.Jets3tProperties).
log4j:WARN Please initialize the log4j system properly.
log4j:WARN See http://logging.apache.org/log4j/1.2/faq.html#noconfig for more info.
12-Dec-2022 09:53:32.779 INFO [main] org.apache.coyote.AbstractProtocol.start Starting ProtocolHandler ["http-nio-8888"]
12-Dec-2022 09:53:32.805 INFO [main] org.apache.catalina.startup.Catalina.start Server startup in [3241] milliseconds


### Run App ###

SLF4J: Failed to load class "org.slf4j.impl.StaticLoggerBinder".
SLF4J: Defaulting to no-operation (NOP) logger implementation
SLF4J: See http://www.slf4j.org/codes.html#StaticLoggerBinder for further details.
[luceedebug] Method 'udfCall' in class 'info/mysite/askaquestion/aaqquestionpool_cfc$cf' became too large after instrumentation (size=70702). luceedebug won't be able to hit breakpoints in, or expose frame information for, this file.
[luceedebug] Method 'udfCall2' in class 'taportal/workplan/services/mixins/workplanscopeservice_cfm$cf' became too large after instrumentation (size=71842). luceedebug won't be able to hit breakpoints in, or expose frame information for, this file.
vagrant@local:~$ docker exec -it $(get_a_container_id_from_service.sh hudx_local_lucee) bash
root@c5b5cae2be75:/usr/local/tomcat# docker-entrypoint.sh
Entrypoint scripts haven't run yet; running...
total 24
drwxr-xr-x 1 root root 4096 Dec 12 09:51 .
drwxr-xr-x 1 root root 4096 Dec 12 10:03 ..
-rwxr-xr-x 1 root root  280 Dec  7 14:44 10_populate_env_from_secrets.sh
-rwxr-xr-x 1 root root  218 Dec  7 14:44 20_set-derived-environment-variables.sh
-rwxr-xr-x 1 root root 2292 Dec  7 14:44 50_populate_templates.sh

/usr/local/bin/docker-entrypoint.sh: running /docker-entrypoint-init.d/10_populate_env_from_secrets.sh

/usr/local/bin/docker-entrypoint.sh: running /docker-entrypoint-init.d/20_set-derived-environment-variables.sh

/usr/local/bin/docker-entrypoint.sh: running /docker-entrypoint-init.d/50_populate_templates.sh
populate_templates.sh: INFO: processing (/tmp/template_renderer/templates/lucee-web.xml.j2) source file
populate_templates.sh: INFO: processing (/tmp/template_renderer/templates/scheduler.xml.j2) source file
populate_templates.sh: INFO: source template (/tmp/template_renderer/templates/scheduler.xml.j2) not found
populate_templates.sh: INFO: processing (/tmp/template_renderer/templates/fusionreactor/reactor.conf.j2) source file
populate_templates.sh: INFO: processing (/tmp/template_renderer/templates/fusionreactor/reports_configuration.config.j2) source file
populate_templates.sh: INFO: processing (/tmp/template_renderer/templates/fusionreactor/ProfileSamplerConfig.config) source file

Using CATALINA_BASE:   /usr/local/tomcat
Using CATALINA_HOME:   /usr/local/tomcat
Using CATALINA_TMPDIR: /usr/local/tomcat/temp
Using JRE_HOME:        /usr/local/openjdk-11
Using CLASSPATH:       /usr/local/tomcat/bin/bootstrap.jar:/usr/local/tomcat/bin/tomcat-juli.jar
Using CATALINA_OPTS:
NOTE: Picked up JDK_JAVA_OPTIONS:  --add-opens=java.base/java.lang=ALL-UNNAMED --add-opens=java.base/java.io=ALL-UNNAMED --add-opens=java.rmi/sun.rmi.transport=ALL-UNNAMED
Listening for transport dt_socket at address: 9999
[luceedebug] agent premain complete
12-Dec-2022 10:05:46.980 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log Server version name:   Apache Tomcat/9.0.39
12-Dec-2022 10:05:46.988 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log Server built:          Oct 6 2020 14:11:46 UTC
12-Dec-2022 10:05:46.989 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log Server version number: 9.0.39.0
12-Dec-2022 10:05:46.990 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log OS Name:               Linux
12-Dec-2022 10:05:46.990 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log OS Version:            4.15.0-200-generic
12-Dec-2022 10:05:46.992 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log Architecture:          amd64
12-Dec-2022 10:05:46.992 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log Java Home:             /usr/local/openjdk-11
12-Dec-2022 10:05:46.993 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log JVM Version:           11.0.9+11
12-Dec-2022 10:05:46.995 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log JVM Vendor:            Oracle Corporation
12-Dec-2022 10:05:46.995 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log CATALINA_BASE:         /usr/local/tomcat
12-Dec-2022 10:05:46.996 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log CATALINA_HOME:         /usr/local/tomcat
12-Dec-2022 10:05:47.032 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log Command line argument: --add-opens=java.base/java.lang=ALL-UNNAMED
12-Dec-2022 10:05:47.033 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log Command line argument: --add-opens=java.base/java.io=ALL-UNNAMED
12-Dec-2022 10:05:47.033 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log Command line argument: --add-opens=java.rmi/sun.rmi.transport=ALL-UNNAMED
12-Dec-2022 10:05:47.033 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log Command line argument: -Djava.util.logging.config.file=/usr/local/tomcat/conf/logging.properties
12-Dec-2022 10:05:47.034 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log Command line argument: -Djava.util.logging.manager=org.apache.juli.ClassLoaderLogManager
12-Dec-2022 10:05:47.034 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log Command line argument: -Xms512m
12-Dec-2022 10:05:47.034 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log Command line argument: -Xmx4096m
12-Dec-2022 10:05:47.035 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log Command line argument: -Djava.security.egd=file:/dev/./urandom
12-Dec-2022 10:05:47.035 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log Command line argument: -agentlib:jdwp=transport=dt_socket,server=y,suspend=n,address=localhost:9999
12-Dec-2022 10:05:47.035 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log Command line argument: -javaagent:/app/lib/luceedebug/luceedebug.jar=jdwpHost=localhost,jdwpPort=9999,cfHost=localhost,cfPort=10000,jarPath=/app/lib/luceedebug/luceedebug.jar
12-Dec-2022 10:05:47.036 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log Command line argument: -Djdk.tls.ephemeralDHKeySize=2048
12-Dec-2022 10:05:47.036 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log Command line argument: -Djava.protocol.handler.pkgs=org.apache.catalina.webresources
12-Dec-2022 10:05:47.036 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log Command line argument: -Dorg.apache.catalina.security.SecurityListener.UMASK=0027
12-Dec-2022 10:05:47.037 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log Command line argument: -Dignore.endorsed.dirs=
12-Dec-2022 10:05:47.037 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log Command line argument: -Dcatalina.base=/usr/local/tomcat
12-Dec-2022 10:05:47.037 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log Command line argument: -Dcatalina.home=/usr/local/tomcat
12-Dec-2022 10:05:47.038 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log Command line argument: -Djava.io.tmpdir=/usr/local/tomcat/temp
12-Dec-2022 10:05:47.047 INFO [main] org.apache.catalina.core.AprLifecycleListener.lifecycleEvent Loaded Apache Tomcat Native library [1.2.25] using APR version [1.6.5].
12-Dec-2022 10:05:47.048 INFO [main] org.apache.catalina.core.AprLifecycleListener.lifecycleEvent APR capabilities: IPv6 [true], sendfile [true], accept filters [false], random [true].
12-Dec-2022 10:05:47.049 INFO [main] org.apache.catalina.core.AprLifecycleListener.lifecycleEvent APR/OpenSSL configuration: useAprConnector [false], useOpenSSL [true]
12-Dec-2022 10:05:47.059 INFO [main] org.apache.catalina.core.AprLifecycleListener.initializeSSL OpenSSL successfully initialized [OpenSSL 1.1.1d  10 Sep 2019]
12-Dec-2022 10:05:47.614 INFO [main] org.apache.coyote.AbstractProtocol.init Initializing ProtocolHandler ["http-nio-8888"]
12-Dec-2022 10:05:47.699 INFO [main] org.apache.catalina.startup.Catalina.load Server initialization in [1216] milliseconds
12-Dec-2022 10:05:47.790 INFO [main] org.apache.catalina.core.StandardService.startInternal Starting service [Catalina]
12-Dec-2022 10:05:47.791 INFO [main] org.apache.catalina.core.StandardEngine.startInternal Starting Servlet engine: [Apache Tomcat/9.0.39]
WARNING: An illegal reflective access operation has occurred
WARNING: Illegal reflective access by org.apache.felix.framework.ext.ClassPathExtenderFactory$DefaultClassLoaderExtender (file:/usr/local/tomcat/lucee/lucee.jar) to method java.net.URLClassLoader.addURL(java.net.URL)
WARNING: Please consider reporting this to the maintainers of org.apache.felix.framework.ext.ClassPathExtenderFactory$DefaultClassLoaderExtender
WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
WARNING: All illegal access operations will be denied in a future release
log4j:WARN No appenders could be found for logger (org.jets3t.service.Jets3tProperties).
log4j:WARN Please initialize the log4j system properly.
log4j:WARN See http://logging.apache.org/log4j/1.2/faq.html#noconfig for more info.
12-Dec-2022 10:05:51.154 INFO [main] org.apache.coyote.AbstractProtocol.start Starting ProtocolHandler ["http-nio-8888"]
12-Dec-2022 10:05:51.222 INFO [main] org.apache.catalina.startup.Catalina.start Server startup in [3522] milliseconds


### Run App ###

SLF4J: Failed to load class "org.slf4j.impl.StaticLoggerBinder".
SLF4J: Defaulting to no-operation (NOP) logger implementation
SLF4J: See http://www.slf4j.org/codes.html#StaticLoggerBinder for further details.
[luceedebug] Method 'udfCall' in class 'info/mysite/askaquestion/aaqquestionpool_cfc$cf' became too large after instrumentation (size=70702). luceedebug won't be able to hit breakpoints in, or expose frame information for, this file.
[luceedebug] Method 'udfCall2' in class 'taportal/workplan/services/mixins/workplanscopeservice_cfm$cf' became too large after instrumentation (size=71842). luceedebug won't be able to hit breakpoints in, or expose frame information for, this file.
```

**Comment by jamiejackson** [comment]
Maybe you can make sense of this. These two probes are within the same Lucee server instance (server hadn't restarted):

```
vagrant@local:~$ docker exec -it $(get_a_container_id_from_service.sh hudx_local_lucee) nmap localhost -p 8888,9999,10000
Starting Nmap 7.70 ( https://nmap.org ) at 2022-12-12 10:29 CST
Nmap scan report for localhost (127.0.0.1)
Host is up (0.000043s latency).
Other addresses for localhost (not scanned): ::1

PORT      STATE  SERVICE
8888/tcp  open   sun-answerbook
9999/tcp  open   abyss
10000/tcp closed snet-sensor-mgmt

Nmap done: 1 IP address (1 host up) scanned in 0.34 seconds
vagrant@local:~$ docker exec -it $(get_a_container_id_from_service.sh hudx_local_lucee) nmap localhost -p 8888,9999,10000
Starting Nmap 7.70 ( https://nmap.org ) at 2022-12-12 10:30 CST
Nmap scan report for localhost (127.0.0.1)
Host is up (0.000059s latency).
Other addresses for localhost (not scanned): ::1

PORT      STATE  SERVICE
8888/tcp  open   sun-answerbook
9999/tcp  closed abyss
10000/tcp open   snet-sensor-mgmt

Nmap done: 1 IP address (1 host up) scanned in 0.28 seconds
```

It looks like 9999 and 10000 "took turns" being open.

**Comment by softwareCobbler** [comment]
![image](https://user-images.githubusercontent.com/18746216/207201414-458d2d3e-e552-46ef-973f-b8ce2ad52356.png)

**Comment by jamiejackson** [comment]
I was able to connect! I'm not sure I have established the minimal series (or even _a_ reliable series) of steps to get it working.

Notes to self to pick up later:

* `cfHost` _must_ be `0.0.0.0` in a docker container. The readme alludes to this but it needs to be stated with more certainty.
* The note about priming the server was helpful, I think. Need to check whether one needs to prime before attempting a connection. I think so.
* Check whether I need to prime with a simple application (a subdirectory with just an `Application.cfc` and an `index.cfm`), then connect, then access the actual, large application.
